packages/insomnia-app/app/ui/components/settings/import-export.tsx

```diff
@@ -1,98 +1,111 @@
-import React, { PureComponent } from 'react';
-import { autoBindMethodsForReact } from 'class-autobind-decorator';
-import { AUTOBIND_CFG } from '../../../common/constants';
+import React, { FC, useCallback } from 'react';
 import { Dropdown, DropdownButton, DropdownDivider, DropdownItem } from '../base/dropdown';
 import Link from '../base/link';
-import { showPrompt } from '../modals/index';
+import { showModal, showPrompt } from '../modals/index';
 import { docsImportExport } from '../../../common/documentation';
 import { strings } from '../../../common/strings';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectActiveSpaceName, selectActiveWorkspace } from '../../redux/selectors';
+import ExportRequestsModal from '../modals/export-requests-modal';
+import { exportAllToFile, importClipBoard, importFile, importUri } from '../../redux/modules/global';
+import { isDesign } from '../../../models/helpers/is-model';
 
 interface Props {
-  handleImportFile: Function;
-  handleImportClipBoard: Function;
-  handleImportUri: Function;
-  handleExportAll: Function;
-  handleShowExportRequestsModal: Function;
+  hideSettingsModal: () => void;
 }
 
-@autoBindMethodsForReact(AUTOBIND_CFG)
-class ImportExport extends PureComponent<Props> {
-  _handleImportUri() {
+export const ImportExport: FC<Props> = ({ hideSettingsModal }) => {
+  const dispatch = useDispatch();
+  const spaceName = useSelector(selectActiveSpaceName);
+  const activeWorkspace = useSelector(selectActiveWorkspace);
+  const activeResourceScope = isDesign(activeWorkspace) ? strings.document : strings.collection;
+
+  const handleImportUri = useCallback(() => {
     const lastUsedImportUri = window.localStorage.getItem('insomnia.lastUsedImportUri');
-    const promptOptions = {
+    const defaultValue = lastUsedImportUri ? { defaultValue: lastUsedImportUri } : {};
+    showPrompt({
       title: 'Import Data from URL',
       submitName: 'Fetch and Import',
       label: 'URL',
       placeholder: 'https://website.com/insomnia-import.json',
-      onComplete: uri => {
+      onComplete: async (uri: string) => {
         window.localStorage.setItem('insomnia.lastUsedImportUri', uri);
-        this.props.handleImportUri(uri);
+        await importUri(activeWorkspace._id, uri)(dispatch);
```

Comment:
Same thing here as per previous feedback

```suggestion
        await dispatch(importUri(activeWorkspace._id))
```

Reply:
I'm kinda confused as to why this looked wrong to you on GitHub because this was updated in 8a9b5a8af7fd19741275505499f8d29ecb62ef28 which was pushed 7 hours before your comment here

Reply:
Yeah I'm not sure either 🤔 maybe some sort of caching issue? Ah well